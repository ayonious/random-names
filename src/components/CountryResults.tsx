import * as React from "react";
import Paper from "@material-ui/core/Paper";
import { database } from "../resources/Database";
import { CustomButton } from "./Utils";

interface Props {
  country: string;
}

interface State {
  currentName: string;
}

export default class CountryReults extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.handleShuffle = this.handleShuffle.bind(this);
    this.state = { currentName: "" };
  }

  handleShuffle() {
    const index = Math.floor(
      Math.random() * Object.keys(database[this.props.country].names).length
    );
    const name = database[this.props.country].names[index];
    this.setState({ currentName: name });
  }

  render() {
    const style = {
      height: 50,
      width: 150,
      margin: "0 auto",
      fontSize: "xx-large",
      contentAlign: "center"
    };
    return (
      <div>
        <br />
        <div style={style}>
          <Paper square={true}>{this.state.currentName}</Paper>
        </div>
        <br />
        <div>
          <CustomButton onClick={this.handleShuffle} title={"Shuffle"} />{" "}
        </div>
      </div>
    );
  }
}
