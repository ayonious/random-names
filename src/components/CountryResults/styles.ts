import styled from 'styled-components';

export const CardWrapper = styled.div`
  height: 50px;
  width: 170px;
  margin: 0 auto;
  font-size: xx-large;
  text-align: center;
`;
