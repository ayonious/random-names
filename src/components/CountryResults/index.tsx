import Card from '@material-ui/core/Card';
import React, { useState } from 'react';

import { database } from '../../resources/Database';
import { CardWrapper } from './styles';
import CustomButton from '../CustomButton';

interface Props {
  country: string;
}

const CountryResults = (props: Props) => {
  const [currentName, changeName] = useState('');

  const handleShuffle = () => {
    const index = Math.floor(
      Math.random() * Object.keys(database[props.country].names).length
    );
    const name = database[props.country].names[index];
    changeName(name);
  };

  return (
    <div>
      <br />
      <CardWrapper>
        <Card square={true}> {currentName} </Card>
      </CardWrapper>
      <br />
      <div>
        <CustomButton onClick={handleShuffle} title={'Shuffle'} />{' '}
      </div>
    </div>
  );
};

export default CountryResults;
