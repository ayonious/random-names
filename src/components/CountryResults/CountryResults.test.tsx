/**
 * @jest-environment jsdom
 */
import { expect } from '@jest/globals';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import CountryResults from '.';
import * as SwitzerlandData from '../../resources/countryInput/Switzerland.json';

it('CountryResults renders', () => {
  const { container } = render(<CountryResults country="Germany" />);
  expect(container).toMatchSnapshot();
});

it('CountryResults after click renders', () => {
  render(<CountryResults country="Switzerland" />);

  const button = screen.getByTestId('shuffle-button');
  const nameOnCardBeforeClick = screen.getByTestId('name-card').textContent;

  expect(nameOnCardBeforeClick).toBe('');

  fireEvent.click(button);

  const nameOnCardAfterClick = screen.getByTestId('name-card').textContent;
  expect(SwitzerlandData.names).toContain(nameOnCardAfterClick);
});
