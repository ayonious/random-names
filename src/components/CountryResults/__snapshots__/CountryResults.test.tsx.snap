// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryResults renders 1`] = `
<div>
  <div>
    <br />
    <div
      class="sc-aYaIB ifiJgs"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 MuiCard-root css-1vqzrnv-MuiPaper-root-MuiCard-root"
        data-testid="name-card"
      />
    </div>
    <br />
    <div>
      <div
        class="sc-gEvDqW jlhBWw"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="shuffle-button"
          tabindex="0"
          type="button"
        >
          Shuffle
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
